### PR TITLE
Fix sui token balances

### LINKED ISSUE #95: Wallet manager not exporting SUI Balances

### PR DESCRIPTION
**Pull request by mat1asm**
Closes #95 
Screenshot from 2023-08-30 15-52-34

### CODE REVIEW

**Review by iturricf — APPROVED** [review]
LGTM!

Just one question, if we pass a known token address instead of the symbol will it recognize it anyway? There's no need to implement that now if that's not the way it's working. We can add it to the backlog perhaps.

**Comment by mat1asm** [comment]
> LGTM!
> 
> Just one question, if we pass a known token address instead of the symbol will it recognize it anyway? There's no need to implement that now if that's not the way it's working. We can add it to the backlog perhaps.

@iturricf  made a few changes to allow for valid token addresses

**Review by iturricf — APPROVED** [review]
Awesome!

File: src/wallets/sui/index.ts [hunk 1]
```diff
@@ -105,13 +110,12 @@ export class SuiWalletToolbox extends WalletToolbox {
   }
 
   public parseTokensConfig(tokens: string[], failOnInvalidTokens: boolean): string[] {
-    const knownTokens = this.getKnownTokens();
     const validTokens: string[] = [];
     for (const token of tokens) {
       if (this.isValidNativeTokenAddress(token)) {
         validTokens.push(token);
       } else if (this.isKnownToken(token)) {
```
Comment (abhidtu2014) [inline src/wallets/sui/index.ts:117]: May be we can update the if condition as well:

```suggestion
     const knownTokens = this.getKnownTokens();
     ...
      } else if (token.toUpperCase() in knownTokens) {
        validTokens.push(knowTokens[token.toUpperCase()]);
      }
```

**Review by abhidtu2014 — APPROVED** [review]

File: src/wallets/sui/index.ts (hunk 1 above)
Comment (abhidtu2014) [reply src/wallets/sui/index.ts:117]: Cool, looks good 👍

**Pull request merged**